bmc/redfish.go

```diff
@@ -342,14 +371,29 @@ func (r *RedfishBMC) checkBiosAttributes(attrs map[string]string) (reset bool, e
 	return
 }
 
-func (r *RedfishBMC) GetStorages(systemUUID string) ([]Storage, error) {
-	system, err := r.getSystemByUUID(systemUUID)
+func (r *RedfishBMC) GetStorages(ctx context.Context, systemUUID string) ([]Storage, error) {
+	system, err := r.getSystemByUUID(ctx, systemUUID)
 	if err != nil {
 		return nil, err
 	}
-	systemStorage, err := system.Storage()
+	var systemStorage []*redfish.Storage
+	err = wait.PollUntilContextTimeout(
+		ctx,
+		r.options.ResourcePollingInterval,
+		r.options.ResourcePollingTimeout,
+		true,
+		func(ctx context.Context) (bool, error) {
+			if ctx.Err() != nil {
+				return false, ctx.Err()
+			}
```

Comment:
Not necessary I think, as `PollUntilContextTimeout` checks for context cancellation.

Reply:
Yes, you don't necessarily need the `ctx.Err()` check inside the polling function because `wait.PollUntilContextTimeout` itself already handles context cancellation and timeout. Including it may be redundant.